js/viewer/annotation/text.js

```diff
@@ -0,0 +1,425 @@
+function Text(attrs) {
+
+    attrs = attrs ? attrs : {};
+    Base.call(this, attrs);
+
+    this._tag = "text";
+    this._attrs["x"] = (attrs["x"] / 2);
+    this._attrs["y"] = (attrs["y"] / 2);
+    this._attrs["stroke"] = attrs["stroke"] || "red";
+
+    // Calculate the ratio for the mapping from image to screen size
+    var viewerParent = this.parent.parent.parent;
+    var containerSize = viewerParent.osd.viewport.containerSize;
+    this._ratio = Math.max(attrs["x"]/containerSize.x, attrs["y"]/containerSize.y);
+
+    this._attrs["font-size"] = attrs["font-size"] || this._ratio * 14;
+    this._attrs["font-weight"] = attrs["font-weight"] || 400;
+    this.svg = null;
+
+    // Foreign object that contains the annotation input and transformed text p tag
+    this.foreignObj = null;
+    this.setForeignObj = function(obj) {
+        this.foreignObj = obj;
+    }
+    this.getForeignObj = function() {
+        return this.foreignObj;
+    }
+}
+
+Text.prototype = Object.create(Base.prototype);
+Text.prototype.constructor = Text;
+
+/**
+ * This function is used to place the annotation at the desired position when user clicks on the OSD canvas
+ * @param {Array} point - The x and y coordinates of the point are used to position the annotation
+ */
+Text.prototype.positionAnnotation = function (point) {
+    var _self = this;
+    var updateAttrs = {
+        x: point.x,
+        y: point.y,
+    };
+
+    _self.setAttributesByJSON(updateAttrs);
+    _self.renderSVG();
+};
+      
+
+
+/**
+ * Function to transform the <textarea> input to <p> tag when annotation tool is switched.
+ */
+Text.prototype.transform = function () {
+
+    // Transform only if the input is not empty
+    if(textInput.value && textInput.value.trim() != "")
+    {
+        var pText = this.createPTag(textInput);
+        var foreignObj = this.getForeignObj();
+        var div = foreignObj.childNodes[0];
+        var h = div.style.top;
+        var w = div.style.left;
+        foreignObj.style.width = pText.style.width;
+        foreignObj.style.height = pText.style.height;
+        var divPadding = window.getComputedStyle(div, null).getPropertyValue('padding');
+        divPadding = parseInt(divPadding.slice(0, divPadding.length - 2)) || 0;
+        var divBorder = window.getComputedStyle(div, null).getPropertyValue('border').split(" ")[0];
+        divBorder = parseInt(divBorder.slice(0, divBorder.length - 2)) || 0;
+        if(h){
+            h = parseInt(h.slice(0, h.length - 2))
+        }
+        else{
+            h = 0;
+        }
+        if(w){
+            w = parseInt(w.slice(0, w.length - 2))
+        }
+        else{
+            w = 0;
+        }
+        foreignObj.setAttribute("x", this._attrs["x"] + w + divPadding + divBorder);
```

Comment:
Commented in the code. This is done to get the actual position the text should be placed at after transformation. Since, before the transformation, the position is different for the start of the annotation and the actual place the text is placed at due to the draggable div and its padding. So, we calculate the new position of only the text in the p tag by taking into account the previous width, padding, and border attributes.

Reply:
Please include the extra explanation that you have here in the code. Use multiline syntax for larger comments like this.

```js
/**
 * your comment...
 */
```

Reply:
Added the extra explanation